packages/typedoc-plugin-appium/lib/converter/comment/index.ts

```
@@ -0,0 +1,182 @@
+  const commentFinders = getFinders(refl);
+
+  /**
+   * The result of running thru all of the comment finders. Each value will have a
+   * {@linkcode CommentSource} corresponding to the finder, and the a `comment` property _if and
+   * only if_ a {@linkcode Comment} was found.
+   */
+  const rawCommentData: SetOptional<CommentData, 'comment'>[] = commentFinders.map(
+    ({getter, commentSource}) => ({
+      comment: getter({refl, comment, knownBuiltinMethods: knownMethods}),
+      commentSource,
+    })
+  );
+
+  /**
+   * The result of filtering out any {@linkcode CommentData} objects which do not have a `comment` property
+   * _or_ which do but the {@linkcode Comment} property does not have any visible components.
+   *
+   * A comment without visible components is defined by TypeDoc as one without a "summary" (or an
+   * empty summary) and one which has no block tags. Block tags, by definition, display something.
+   */
+  const commentData: CommentData[] = rawCommentData.filter(({comment}) =>
+    comment?.hasVisibleComponent()
+  ) as CommentData[];
+
+  // the first summary found; this is where precedence comes in to play
+  const summaryCommentData = commentData.find(({comment}) => comment.summary.length);
+
+  // in this bit we want to pull all block tags from all comments, and see which have content and
+  // which don't.  for each tag name (e.g., `@returns`), prefer the tag with content.  if there is
+  // no such tag, just pick the first one
+  const allBlockTags = commentData.flatMap(({comment}) => comment.blockTags);
+  const tagsByTag = _.groupBy(allBlockTags, 'tag');
+  const finalBlockTags: CommentTag[] = [];
+  for (const tags of Object.values(tagsByTag)) {
+    if (tags.length === 1) {
+      finalBlockTags.push(_.first(tags)!);
+    } else if (tags.length > 1) {
+      // prefer a tag with content
+      const tag = tags.find((t) => t.content.length) ?? tags[0];
+      finalBlockTags.push(tag);
+    } else {
+      // should not happen
```

do we want to throw an error here so if it _does_ happen we know that it happened?

---

(I'll log it)